lib/Logger.js

```diff
@@ -32,20 +36,28 @@ class Logger {
      *                                      the 'level' field is not accounted
      *                                      for)
      *
-     * @param {string} name  - The name of the Logger. It can be found later on
-     *                         in the log entries.
      *
      * @see [Bunyan's documentation]{@link
      * https://github.com/trentm/node-bunyan/blob/master/README.md#streams} for
      * a more detailed description of the streams array configuration.
      *
      * @returns {undefined}
      */
-    constructor(config, name) {
-        this.config = config;
+    constructor(name, config) {
+        this.config = new Config();
+        if (config) {
```

Comment:
Likewise, this condition here is actually _**reinstating**_ the bug we're trying to fix.
Config should not be handled in Logger anymore, it should only be kept as a reference, to access values on-demand, as my original patch offered.